### ISSUE TITLE
"match is unsupported" error message should be improved

### ISSUE THREAD

**Issue by Zamiell** [Perryvw/TypeScriptToLua#1503]
The message should explicitly tell you to use `string.match(foo, regex)`.

**Comment by Z3rio** [comment]
Could you please supply a sample ts snippet to show what the issue is a bit more specifically, thank you :)

**Comment by Zamiell** [comment]
Sorry, I was not clear enough in the original post.

Consider the following code snippet, which is extremely common:

```ts
const string1 = "The quick brown fox jumps over the lazy dog.";
const match = string1.match("brown (\w+) jumps");
```

The playground for this is here:
 https://typescripttolua.github.io/play#code/MYewdgzgLgBNBOBLMBzAjDAvDARAFQAsBTGARwFdFgBrGAI3hAHcwYAzEADxgCtyBbAA4QYIAG5F4MKMRgAbAIYAvAJ4wAJiBQA6HAG4AUKEix+CqMAJY4UJKjTazFggAocDZqxcAdJgGoASl4BYRwAwyA

We can see that it throws the error:
"string.match is unsupported".

But this error message sucks and is misleading, because what it really means is that the `String.match` method is unsupported. The Lua global `string.match` is absolutely supported.

What we really need to do here is to reconfigure the code to look like this:

```ts
const string1 = "The quick brown fox jumps over the lazy dog.";
const match = string.match(string1, "brown (%w+) jumps");
```

So, in conclusion, I think the error message should say:

> The `String.match` method is unsupported. Consider including the Lua types for your environment and then changing the code to `string.match(foo, regex)` instead. Note that the Lua match works differently than the JavaScript/TypeScript match, so for example you would have to change `\w+` to `%w+`.
